packages/@aws-cdk/aws-ecs-patterns/test/fargate/test.scheduled-fargate-task.ts

```diff
@@ -251,4 +251,47 @@ export = {
 
     test.done();
   },
+
+  "Scheduled Fargate Task - with subnetSelection defined"(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'Vpc', {
+      maxAzs: 1,
+      subnetConfiguration: [
+        { name: 'Public', cidrMask: 28, subnetType: ec2.SubnetType.PUBLIC }
+      ],
+    });
+    const cluster = new ecs.Cluster(stack, 'EcsCluster', { vpc });
+
+    new ScheduledFargateTask(stack, 'ScheduledFargateTask', {
+      cluster,
+      scheduledFargateTaskImageOptions: {
+        image: ecs.ContainerImage.fromRegistry('henk'),
+      },
+      subnetSelection: { subnetType: ec2.SubnetType.PUBLIC },
+      schedule: events.Schedule.expression('rate(1 minute)')
+    });
+
+    // THEN
+    expect(stack).to(haveResourceLike('AWS::Events::Rule', {
+      Targets: [
+        {
+          EcsParameters: {
+            NetworkConfiguration: {
+              AwsVpcConfiguration: {
+                AssignPublicIp: 'ENABLED',
+                Subnets: [
+                  {
+                    Ref: 'VpcPublicSubnet1Subnet5C2D37C4'
```

Comment:
This test looks better to me now. Where does this value originate from though?

Reply:
I suspect the "Vpc" prefix comes from the string ID argument supplied on [this line][1] and the word "Public" comes from the `name` property supplied on [this line][2]. I believe the rest is generated by the `Vpc` CDK resource or its sub-resources. The suffix seems to be deterministic, i.e. the same every time the test is run. I believe this behaviour is intentional in the CDK to ensure the same underlying CloudFormation resources are identified each time.

All the other tests which make assertions against the `AWS::Events::Rule` check for a private subnet with reference "VpcPrivateSubnet1Subnet536B997A" which is generated in a similar way, i.e. there is precedent for this. I'm not sure there's a good way to make it clear where the reference value comes from. Do you have any ideas?

As per my previous comment, I'd be happy to try to improve these tests as a separate PR, but I'd be quite keen to get this one merged first so I can start using the new option if that's OK.

[1]: https://github.com/aws/aws-cdk/pull/6624/commits/f8fdc4fca88a59591c396adfb39dd78d7658213d#diff-64a7aca333fa14032084fbd40ac0e30bR258
[2]: https://github.com/aws/aws-cdk/pull/6624/commits/f8fdc4fca88a59591c396adfb39dd78d7658213d#diff-64a7aca333fa14032084fbd40ac0e30bR261

Reply:
> there is precedent for this

I think this is fine then. No need to add a comment or anything.